### ISSUE TITLE
decode 'months since' for 360_day calendars

### ISSUE THREAD

**Issue by rabernat** [Unidata/cftime#68]
The [IRI Data Library](https://iridl.ldeo.columbia.edu) contains tons of datasets with the following time attributes:
```
	float32 T(T) ;
		T:standard_name = time ;
		T:pointwidth = 1.0 ;
		T:long_name = Time ;
		T:calendar = 360 ;
		T:expires = 1538524800 ;
		T:gridtype = 0 ;
		T:units = months since 1960-01-01 ;
```
(edited to correct typo in original post)

([example dataset](https://iridl.ldeo.columbia.edu/SOURCES/.NOAA/.NCEP-NCAR/.CDAS-1/.MONTHLY/.Diagnostic/.surface/.temp/dods))

I would like to be able to open and decode these datasets in xarray, with time decoding handled by cftime.

There are two problems:
1. The calendar is not a valid CF calendar. It should be `360_day`. But that is easy to fix by rewriting the calendar attribute.
1. Even with `calendar = 360_day`, `months` is not considered a valid time unit

However, in a 360-day calendar month==30 days, so this should be valid.

This was discussed over in https://github.com/Unidata/netcdf4-python/issues/434#issuecomment-133083060, where @jswhit commented:
> A pull request allowing months since when calendar is 360_day would be welcome

**Comment by rabernat** [comment]
To clarify, currently in cftime, I get the following:

```python
cftime.num2date(0, 'months since 1960-01-01', '360_day')
```
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-65-4a67652d64b1> in <module>()
----> 1 cftime.num2date(0, 'months since 1960-01-01', '360_day')

cftime/_cftime.pyx in cftime._cftime.num2date()

cftime/_cftime.pyx in cftime._cftime.utime.__init__()

cftime/_cftime.pyx in cftime._cftime._parse_date_and_units()

ValueError: units must be one of 'seconds', 'minutes', 'hours' or 'days' (or singular version of these), got 'months'
```

**Comment by jswhit** [comment]
Created a new pull request #75 which includes both the date2num and num2date implementations, and works with the latest updates to master.

**Comment by jswhit** [comment]
merged pull request #75

**Comment by davidhassell** [comment]
Sorry, I've not been able to look at this for a few days - does this mean that "months since ..." are _always_ decoded as 30 day months when the calendar is "360_day"?
